lib/cachext/breaker.rb

```diff
@@ -0,0 +1,90 @@
+require "thread"
+
+module Cachext
+  class Breaker
+    attr_reader :config
+
+    def initialize(config)
+      @config = config
+    end
+
+    def for(key)
+      WithKey.new(config, key)
+    end
+
+    class WithKey
+      attr_reader :config, :key
+
+      def initialize(config, key)
+        @config = config
+        @key = key
+      end
+
+      def increment_failure
+        redis.pipelined do
+          redis.set key_str(:last_failure), Time.now.to_f
+          redis.incr key_str(:monitor)
+        end
+      end
+
+      def check_health
+        if half_open?
+          if health_check >= config.failure_threshold
```

Comment:
No, I think you got it. The intention is to see how many times we've tried the half-open call. If the service is still failing, we want to try once before opening again. But if the service is good, we make sure we get `failure_threashold` good responses back before closing the circuit.